packages/styletron-react/src/index.js

```diff
@@ -107,6 +114,52 @@ type createStyledOpts = {
   ) => React.ComponentType<any>,
 };
 
+function checkNoopEngine(engine: StandardEngine) {
+  // if no engine provided, we default to no-op, handy for tests
+  // however, print a warning in other envs
+  if (process.env.NODE_ENV !== "test") {
+    engine.noopEngine &&
+      // eslint-disable-next-line no-console
+      console.warn(
+        __DEV__
+          ? `
+Styletron has been switched to a no-op (test) mode.
+
+A Styletron styled component was rendered, but no Styletron engine instance was provided in React context.
+
+Did you forget to provide a Styletron engine instance to React context via using the Styletron provider component?
+
+Note: Providers and Consumers must come from the exact same React.createContext call to work.
+If your app has multiple instances of the "styletron-react" package in your node_module tree,
+your Provider may be coming from a different React.createContext call, which means the styled components
+will not recieve the provided engine instance. This scenario can arise, for example, when using "npm link".
+`
+          : `Styletron Provider is not set up. Defaulting to no-op.`,
+      );
+  }
+}
+
+export function useStyletron() {
+  const styletronEngine: StandardEngine = React.useContext(StyletronContext);
+  const debugEngine = React.useContext(DebugEngineContext);
+  const hydrating = React.useContext(HydrationContext);
+  return [
+    function css(style: StyleObject) {
+      checkNoopEngine(styletronEngine);
```

Comment:
Do you think it is necessary for this check every `css` call?  Or should this be hoisted up a level to when `useStyletron` is called.

Reply:
good point, moved